src/classes/BDI_MigrationMappingUtility.cls

```diff
@@ -32,193 +32,614 @@
+            this.sourceFieldAPIName = sourceFieldMapping.Source_Field_API_Name__c;
+            this.targetFieldAPIName = sourceFieldMapping.Target_Field_API_Name__c;
+            this.targetObjectMappingRecordName = sourceFieldMapping.Target_Object_Mapping__r.DeveloperName;
+            this.targetObjectMappingLabel = sourceFieldMapping.Target_Object_Mapping__r.MasterLabel;
+            this.dataImportObjectName = sourceFieldMapping.Target_Object_Mapping__r.DeveloperName;
+            this.targetObjectAPIName = sourceFieldMapping.Target_Object_Mapping__r.Object_API_Name__c;
+            this.recordName = sourceFieldMapping.DeveloperName;
+            this.isRequired = 'No';
+            this.isDeleted = false;
+        }
 
-    /** @description Default packaged Object Mapping Set metadata record */
-    private static final Data_Import_Object_Mapping_Set__mdt DEFAULT_OBJECT_MAPPING_SET = [SELECT 
-                Id, 
-                DeveloperName, 
-                Label
-        FROM Data_Import_Object_Mapping_Set__mdt
-        WHERE DeveloperName = :DEFAULT_OBJECT_MAPPING_SET_RECORD_NAME];
+        public void setLookupsAndTargetField(Data_Import_Object_Mapping__mdt objectMapping,
+            String fieldMappingSetName,
+            String targetFieldAPIName) {
 
-    /** @description Default packaged Field Mapping Set metadata record */
-    private static final Data_Import_Field_Mapping_Set__mdt DEFAULT_FIELD_MAPPING_SET = [SELECT 
-                Id, 
-                DeveloperName, 
-                Label
-        FROM Data_Import_Field_Mapping_Set__mdt
-        WHERE DeveloperName = :DEFAULT_FIELD_MAPPING_SET_RECORD_NAME];
+            this.targetObjectMappingLabel = objectMapping.Label;
+            this.targetObjectMappingRecordName = objectMapping.DeveloperName;
+            this.dataImportFieldMappingSetRecordName = fieldMappingSetName;
+            if (targetFieldAPIName != null) {
+                this.targetFieldAPIName = targetFieldAPIName;
+            }
+        }
 
-    /**
-     * @return A 10 character unique 'jobId' to use for
-     */
-    private static String getUniqueJobId() {
-        return UserInfo.getUserId().left(15).right(4).toUpperCase() +
-                String.valueOf(Integer.valueOf(math.rint(math.random()*5000000))).right(6);
+        public void setRecordName(String recordName, Boolean isUnique) {
+            if (isUnique) {
+                this.recordName = generateRecordName(this.label, true, this.targetObjectMappingLabel, this.dataImportObjectName);
+            } else {
+                this.recordName = recordName;
+            }
+        }
+
+        public Metadata.CustomMetadata getMetadataRecord() {
+            String mdTypeName = 'Data_Import_Field_Mapping__mdt';
+
+            // Setup custom metadata to be created or updated in the subscriber org.
+            Metadata.CustomMetadata customMd = new Metadata.CustomMetadata();
+            if (this.targetObjectMappingLabel != null) {
```

Comment:
`this.` should be used when assigning values to the inner classes' local member variables inside of the constructor only.  When not assigning values inside the constructor, `this.` can be removed.  This can be applied to the other inner classes also.